sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy

```diff
@@ -0,0 +1,609 @@
+            .setRecordSeparator('\n' as char)
+            .setColumnSeparator(',' as char)
+            .setEscapeChar('\0' as char)
+            .setFieldQuote('\0' as char)
+            .setHeadersPresent(false)
+        uploadCsv(ser, numCopies)
+        def expression = "SELECT * from BlobStorage"
+
+        ByteArrayOutputStream downloadData = new ByteArrayOutputStream()
+        bc.download(downloadData)
+        byte[] downloadedData = downloadData.toByteArray()
+
+        /* Input Stream. */
+        when:
+        InputStream qqStream = bc.openQueryInputStream(expression)
+        byte[] queryData = readFromInputStream(qqStream, downloadedData.length)
+
+        then:
+        notThrown(IOException)
+        for (int j = 0; j < downloadedData.length; j++) {
+            assert queryData[j] == downloadedData[j]
+        }
+
+        /* Output Stream. */
+        when:
+        OutputStream os = new ByteArrayOutputStream()
+        bc.query(os, expression)
+        byte[] osData = os.toByteArray()
+
+        then:
+        notThrown(BlobStorageException)
+        for (int j = 0; j < downloadedData.length; j++) {
+            assert osData[j] == downloadedData[j]
+        }
+
+        // To calculate the size of data being tested = numCopies * 32 bytes
+        where:
+        numCopies | _
+        1         | _ // 32 bytes
+        32        | _ // 1 KB
+        256       | _ // 8 KB
+        400       | _ // 12 ish KB
+        4000      | _ // 125 KB
+    }
+
+    /* Note: Input delimited tested everywhere else. */
+    @Unroll
+    def "Query Input json"() {
+        setup:
+        BlobQueryJsonSerialization ser = new BlobQueryJsonSerialization()
+            .setRecordSeparator(recordSeparator as char)
+        uploadSmallJson(numCopies)
+        def expression = "SELECT * from BlobStorage"
+
+        ByteArrayOutputStream downloadData = new ByteArrayOutputStream()
+        bc.download(downloadData)
+        byte[] downloadedData = downloadData.toByteArray()
+        BlobQueryOptions options = new BlobQueryOptions().setInputSerialization(ser).setOutputSerialization(ser)
+
+        /* Input Stream. */
+        when:
+        InputStream qqStream = bc.openQueryInputStream(expression, options)
+        byte[] queryData = readFromInputStream(qqStream, downloadedData.length)
+
+        then:
+        notThrown(IOException)
+        for (int j = 0; j < downloadedData.length; j++) {
+            assert queryData[j] == downloadedData[j]
+        }
+
+        /* Output Stream. */
+        when:
+        OutputStream os = new ByteArrayOutputStream()
+        bc.queryWithResponse(os, expression, options, null, null)
+        byte[] osData = os.toByteArray()
+
+        then:
+        notThrown(BlobStorageException)
+        for (int j = 0; j < downloadedData.length; j++) {
+            assert osData[j] == downloadedData[j]
+        }
+
+        where:
+        numCopies | recordSeparator || _
+        0         | '\n'            || _
```

Comment:
Seems like the recordSeparator variable can just be a constant.

Reply:
I'll add some tests testing other ones